connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -71,6 +72,15 @@ class SparkConnectPlanner(plan: proto.Relation, session: SparkSession) {
     session.sessionState.sqlParser.parsePlan(sql.getQuery)
   }
 
+  private def transformSample(rel: proto.Sample): LogicalPlan = {
+    Sample(
```

Comment:
We cannot. This is proto and for those primitive type there is no `set/unset`: they have default values.

For example, int32 if it is not set then the default value is 0. Server side does not know if this is set or not set, and 0 is a valid input.

The only solution I can think of is that we wrap each of the primitive type field into a message. However that complicate the proto and I am afraid that we don't want to go to that path.

Reply:
cc @cloud-fan as this could be a question appearing over and over again when working with proto.

Reply:
Alright, I think this is something that we have to keep in mind when dealing with the protos. When we want optionality for scalar types we need to wrap them or properly deal with the zero initialized value.

I think in this case we can probably defer the checking to the analyzer because it will know if the value combination is invalid.

Maybe it's a good thing that we start some kind of developer docs with these caveats?

Reply:
Let me add some comments.

Yeah if we ever find this is a true problem we can start to wrap those (maybe not for all fields but a few problematic ones when it applies).